client/jetpack-cloud/sections/agency-dashboard/sites-overview/index.tsx

```diff
@@ -246,9 +248,20 @@ export default function SitesOverview() {
 	const [ hasDismissedProvisioningNotice, setHasDismissedProvisioningNotice ] =
 		useState< boolean >( false );
 	const isProvisioningSite =
-		'true' === getQueryArg( window.location.href, 'provisioning' ) ||
+		'true' === getQueryArg( window.location.href, QUERY_PARAM_PROVISIONING ) ||
 		( ! isLoadingProvisioningBlogIds && Number( provisioningBlogIds?.length ) > 0 );
 
+	const onDismissProvisioningNotice = () => {
+		setHasDismissedProvisioningNotice( true );
+
+		// Delete query param 'provisioning' from the URL.
+		window.history.replaceState(
```

Comment:
We need to update history directly instead of using `page.redirect` as we do not want the page to reload.